### ISSUE TITLE
Possibility to restrict sync using network settings

### ISSUE THREAD

**Issue by BSG2000** [aluxnimm/outlookcaldavsynchronizer#202]
Possibility to restrict sync using network settings
It would be great to run the sync only if you are in a particular network. The restriction could be based on the WLAN SSID or the IP address. Alternatively, you could test the availability of the CALDAV Sync Server. In environments where the server can only be reached in the company subnet, the sync could be used without constant error messages. That would be great!
DAVDroid for Android uses the SSID for a similar function.

**Comment by aluxnimm** [comment]
not planned atm but should be possible.

**Comment by BSG2000** [comment]
Here is an example of how it was solved in DAVdroid 
[new setting: only sync in specific WiFI (by SSID)](https://gitlab.com/bitfireAT/davdroid/commit/6ffa6fa9a75104aad4a7305ac8715e6f6793dfce)

**Comment by BSG2000** [comment]
Is there any plan to get the possibility to use profiles with special connection requirements, like an caldav Server in the intranet in a company?

**Comment by nertsch** [comment]
> It would be great to run the sync only if you are in a particular network.

Since a computer can be connected to multiple networks this is a little bit unclear.
Do you mean
a) sync only if connected to a particular network  or
b) sync only if the server is accessed over a specific network?

br
Gerhard

**Comment by BSG2000** [comment]
I see use cases for both possibilities:

_a) You are in a specific IP range, which is only used by the corporate network. Your client has, for example, the IP 10.123.321.111 in this subnet. It is also possible to determine the WLAN SSID of the connected network.
**or**
b) Before the sync it is checked if the target is reachable at all and the sync is paused for a longer time and no error message is displayed._



In the end, a solution has to be found that performs the sync when you reach the server (for example via VPN) or you are on site in the company's network. It should be avoided that with every Outlook start the window with the error messages is indicated.

**Comment by aluxnimm** [comment]
For b) it is hard or impossible to distinguish if the target not reachable means you are in a wrong network or the server has problems itself. But in that later case there should be an error message...

**Comment by BSG2000** [comment]
Yes that's right. But I just could not be in the corporate network too. ;-)

My suggestion for b): If the user has no intranet restriction configured -> error message will be displayed. If the user has set the option -> no error message, maybe there is an interval and a warning:
"You have not synced since 7d, Connect to your corporate network to update calendars and contacts."
